[ACL] Improve panel, solve ajax problems and ACL bugs

andrepereiradasilva opened the pull request:
Pull Request for solving several ACL issues.
#### Summary of Changes

This PR makes several changes in the permissions panel and also replicate js improvements in sendtest mail in order to solves bugs and improve php/js code.
###### Bugs/Improvements

1: Ajax js errors (json parse, no content, http status error, etc) are not captched in permission.js
After this PR uses the same checks as sendtestmail.js that where now moved to core.js

2: Session token is not being checked in permission.js
After this PR the session token is checked.

3: Permision js ajax call is returned in html mimetype
After this PR the ajax call is returned in json mimetype.

4: Some messages returned by permission and sentestmail js ajax calls are returned in HTML format.
After this PR all messages are returned in Json.

5: The js messages in sendtestmail and permissions.js are not cleaned on change.
After this PR they are cleaned everytime you change.

6: Permission js uses a custom html message system
After this PR all message use the Joomla API message system.

7: A super user cannot remove super user permission from other groups that his not into.
After this PR this is possible.

8: When a super user permission is changed or a permission of a group with child groups is changed, the user is not informed that we needs to refresh the page to recalculate permissions.
After this PR the user is notified.

9: A user that is not Super User can change a Super User group permissions.
After this PR this is not possible.

10: A user that is not Super User can change change its own's groups ot it's parent permissions.
After this PR this is not possible.

11: We don't have calculate permissions column in com_config "Public" (root) group.
After this PR we have the calculated permission here too.

12: Changing a permission with "Not Allowed (Locked)" to "Allowed" or "Denied" changes it's calculate permission (it should stay at "Not Allowed (Locked)"). 
After this PR the calculated permissions always stays "Not Allowed (Locked)" in this cases.

13: If you change a "Inherited" permission with a calculated permission of "Not Allowed" to "Allowed" and then back to "Inherited", the calculated status is changed to "Allowed" (it should be "Not Allowed  (Inherited)").
This PR solves that.

14: There are some "Conflict" messages when you have a parent group with "Denied" and some child group with "Allowed". This should be only "Not Allowed (Locked)"
This PR solves that.

There can be more bugs/improvements that i don't remember...
##### Scenarios
###### Calculated Permissions Scenarios

If only "Inherited" across the group tree or group in global config:
- "Inherited" -> "Not Allowed (Inherited)" (red).
- "Allowed" -> "Allowed" (green).
- "Denied" -> "Not Allowed" (red).

If any "Denied" across the group tree or group in global config:
- "Inherited" -> "Not Allowed (Locked)" (red with lock).
- "Allowed" -> "Not Allowed (Locked)" (red with lock).
- "Denied" -> "Not Allowed (Locked)" (red with lock).

If no "Denied" and a "Allowed" across the group tree or group in global config:
- "Inherited" -> "Allowed (Inherited)" (green).
- "Allowed" -> "Allowed" (green).
- "Denied" -> "Not Allowed" (red).

_Special case_: If group has super User permissions all values in that group and child groups should be "Allowed (Super User)"  (green with lock).
###### Error/Notices Scenarios

Can't change permissions scenarios:
- Session token is not valid.
- Database query/update errors.
- Any error while checking or storing asset.
- Current user does not have permissions to change ACL for the component.
- Current user is Super User and is trying to remove his own Super User permissions.
- Current user is not Super User and is trying to change a Super User group permissions.
- Current user is not Super User and is trying to change its own's groups permissions.
- Current user is not Super User and is trying to change its own's groups parent permissions.

Ask user to reload the page scenarios:
- User changed permissions of a group with child groups.
- User changed group permissions to Super User.
#### Testing Instructions
- Use latest staging.
- Check all "Bugs/Improvements".
- Apply patch
- Go to global configuration
- Clean browser cache.
- Check all "Bugs/Improvements" are solved.
- Check all scenarios described in the "Scenarios".
- Do the tests two times. With your super user and other user with less permissions (easier if you open a session for that user in a private browser window).

andrepereiradasilva committed:
ajax js error messages

andrepereiradasilva committed:
add the js error messages to views

andrepereiradasilva committed:
send test mail should send always json

andrepereiradasilva committed:
simplify sendmailjs

andrepereiradasilva committed:
centralzie ajax errors message sin core

andrepereiradasilva committed:
message, not sucess

andrepereiradasilva committed:
don't render messages if null

andrepereiradasilva committed:
better json/javascript handling

andrepereiradasilva committed:
remove extra debug

andrepereiradasilva committed:
corrrect admin check by component

andrepereiradasilva committed:
app not this->app

andrepereiradasilva committed:
ups

andrepereiradasilva committed:
ups ... again

andrepereiradasilva committed:
remove messages if they exist

andrepereiradasilva committed:
remov messages in sendmail too

andrepereiradasilva committed:
to do and superuser doesn't change

andrepereiradasilva committed:
a super user can demote other groups from being a super user

andrepereiradasilva committed:
check if current group has super user permissions

andrepereiradasilva committed:
language var doesn't exist

andrepereiradasilva committed:
isUserSuperUser

andrepereiradasilva committed:
we can have child groups ...

andrepereiradasilva committed:
remvoe extra spaces

andrepereiradasilva committed:
if user is not is superuser group, cannot change a superuser group

andrepereiradasilva committed:
a super user can change other groups

andrepereiradasilva committed:
improve comments

andrepereiradasilva committed:
improve messages

andrepereiradasilva committed:
not Locked, should be Inherited

andrepereiradasilva committed:
better permissions

andrepereiradasilva committed:
alert recalculate

andrepereiradasilva committed:
solves conflicts 1

andrepereiradasilva committed:
Merge remote-tracking branch 'refs/remotes/joomla/staging' into improve-ajax-js-errors

andrepereiradasilva committed:
solve conflicts final

andrepereiradasilva committed:
a user (not super user) can't change it's own group(s) permissions

andrepereiradasilva committed:
better comments and messages

andrepereiradasilva committed:
add language vars

andrepereiradasilva committed:
show calculated permissions also on Public group in global config!

andrepereiradasilva committed:
ups

andrepereiradasilva committed:
check session token!

andrepereiradasilva committed:
do not mark as confict.

andrepereiradasilva committed:
replace conflict by not allowed (inhreited)

andrepereiradasilva committed:
don't allow to change permissions of his own parent groups (except if it's a super user)

andrepereiradasilva committed:
cs

andrepereiradasilva commented:
pinging @infograf768 @roland-d

dgt41 commented on libraries/joomla/form/fields/rules.php:
```diff
@@ -144,6 +144,10 @@ protected function getInput()
 
+		$permissionsUri = 'index.php?option=com_config&task=config.store&format=json&' . JSession::getFormToken() . '=1';
+
+		JFactory::getDocument()->addScriptDeclaration('var permission_url = "' . addslashes(JUri::base()) . $permissionsUri . '";');
```
@andrepereiradasilva can you pass the permission_url as a data-attribute to the table tag? Then collect it in the permissions.js. Dropping inline scripts is always a good thing!

andrepereiradasilva replied:
yeah, i know. i've changed so many things that have been lazy on this.
i will change it.

andrepereiradasilva replied:
done. Also done some other otimizations.

andrepereiradasilva committed:
remove inline urls, minor otimizations

andrepereiradasilva committed:
"Not Allowed (Inherited)" should be red since is denied

andrepereiradasilva committed:
correct js comments

infograf768 commented:
Please add permissions.min.js to delete in script.php

infograf768 commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

roland-d commented:
Why remove the min.js file?

infograf768 commented:
@roland-d 
Because Andre replaced it by an uncompressed.js one in this patch
He could also simply keep the min.js name.

roland-d commented:
### Testing

1. Super user cannot change it's own permission :white_check_mark: 
2. Parent group has option set to Denied, Child is set to Denied the message is **Not allowed (Inherited)** but on page refresh it says **Denied** :red_circle: 
3. If a component has no entry in the asset table, storing a permission creates an invalid asset entry :red_circle: 
![invalid_asset_entry](https://cloud.githubusercontent.com/assets/359377/15989910/d15a22dc-3085-11e6-8ac9-8becf48cf39b.png)
4. When I change a permission in a parent group I get this message:
 ```
Notice 
Permissions changed in a group with child groups. Save or reload to recalculate the child groups  permissions.
 ```
The message is OK, but every other change I make is causing the screen to jump up and down. Would be nicer if the screen wouldn't jump up and down. :red_circle: 
5. The correct calculated value is returned for setting Allowed/Denied/Inherited :white_check_mark: 
6. User cannot change the permission of it's parent or own group :white_check_mark: 
7. User cannot change a super user permission :white_check_mark: 

@infograf768 
> Because Andre replaced it by an uncompressed.js one in this patch
I see, usually we have a minified version and a regular version as these are loaded as needed by Joomla depending on debug status. No worries, was just wondering.

andrepereiradasilva committed:
add permissions.min.js to remove script

infograf768 commented:
>Parent group has option set to Denied, Child is set to Denied the message is Not allowed (Inherited) but on page refresh it says Denied 🔴 

I can't  reproduce this (Tested on a single article): It remains on Not Allowed (inherited) here.✅

>When I change a permission in a parent group I get this message: Notice Permissions changed in a group with child groups. Save or reload to recalculate the child groups permissions. The message is OK, but every other change I make is causing the screen to jump up and down. Would be nicer if the screen wouldn't jump up and down.

Agree, but I can live with it. :smiley:

>If a component has no entry in the asset table, storing a permission creates an invalid asset entry 🔴 invalid_asset_entry

To test this, I deleted all Permissions in com_banners, just left `{}`. I had no other asset for com_banners.
Then I edited the permissions of the banners component, forbidding create, delete, edit and edit state to Manager (group id 6). 
Then I saved, got the notice, and `Not Allowed (Inherited)` showing OK in children groups.

Looking at db, I get
![screen shot 2016-06-12 at 17 17 31](https://cloud.githubusercontent.com/assets/869724/15991971/97b7f310-30c1-11e6-9522-babee73e5c36.png)

A com_banners category which had no assets is now present in the assets table as
`com_banners.category.3` with the correct parent_id
![screen shot 2016-06-12 at 17 20 16](https://cloud.githubusercontent.com/assets/869724/15991992/fac8d2c6-30c1-11e6-9041-8130c657a7ea.png)

Parents and level look OK to me:

![screen shot 2016-06-12 at 17 24 43](https://cloud.githubusercontent.com/assets/869724/15992028/a9a455d6-30c2-11e6-9c2a-e26e37f1719f.png)

infograf768 commented:
@roland-d  just saw one of your com_banners has an 's' too much

roland-d commented:
@infograf768 It doesn't have an s too much, I have done that on purpose. You misunderstood the test I did. The test is that if you install a component which has no entry (nothing) in the assets table, the entry will be created but incorrect. You tested an empty param column instead.

> Agree, but I can live with it. :smiley:

I can't. I find it very annoying.

> I can't reproduce this (Tested on a single article)

I will test this again another time, let's see what others find.

andrepereiradasilva commented:
@roland-d thanks for testing.

> When I change a permission in a parent group I get this message:  Notice Permissions changed in a group with child groups. Save or reload to recalculate the child groups permissions.  The message is OK, but every other change I make is causing the screen to jump up and down. Would be nicer if the screen wouldn't jump up and down. :red_circle:

Can be easily changed.
But please understand i made that to make sure the user always see the message (mobile for instance). Also remember you're testing and using a LOT of times. IMHO a regular user will not change it so many times.

I will check your other comments.

brianteeman commented:
It's good I see the messages. Usually I don't see them at all in my screen
size

andrepereiradasilva committed:
corrects 2 bugs

- super user should always be Allowed (Super User), even if with a
parent group was an explicity Denied.
- Parent group has option set to Denied, Child is set to Denied the
message is Not allowed (Inherited) but on page refresh it says Denied

andrepereiradasilva commented:
> Parent group has option set to Denied, Child is set to Denied the message is Not allowed (Inherited) but on page refresh it says Denied :red_circle:

Solved.

Also solved another bug. When parent group is set to denied and a child group is Suer user before last commit the calculated permission was "Not Allowed (Inhreited)".

andrepereiradasilva committed:
cs

andrepereiradasilva commented:
@roland-d 
> 3: If a component has no entry in the asset table, storing a permission creates an invalid asset entry :red_circle: invalid_asset_entry

This already happens before the patch right?

andrepereiradasilva committed:
simplify if statements

infograf768 commented:
@roland-d 
Before last changes from @andrepereiradasilva (which I will test now), I have no issue when installing new component.
Exemple with patchtester
![screen shot 2016-06-13 at 07 29 55](https://cloud.githubusercontent.com/assets/869724/15997856/013ee736-3139-11e6-9828-95e79085b370.png)

roland-d commented:
@andrepereiradasilva Yes, this already happens before this patch.
@infograf768 I didn't say I have the issue but was thinking in case you have an issue with your asset table.

infograf768 commented:
>Parent group has option set to Denied, Child is set to Denied the message is Not allowed (Inherited) but on page refresh it says Denied 🔴

>Solved.

That WAS working before. I now systematically get Not Allowed (inherited)

infograf768 commented:
>Also solved another bug. When parent group is set to denied and a child group is Suer user before last commit the calculated permission was "Not Allowed (Inhreited)" (it should be Allowed "(Super User)").

OK here.

andrepereiradasilva commented:
> That WAS working before. I now systematically get Not Allowed (inherited)

And that is how it's supposed to be right?

> If any "Denied" across the group tree:
> 
> "Inherited" -> "Not Allowed (Inherited)" (red with lock).
> "Allowed" -> "Not Allowed (Inherited)" (red with lock).
> "Denied" -> "Not Allowed (Inherited)" (red with lock).

(see PR introduction)

andrepereiradasilva commented:
> @andrepereiradasilva Yes, this already happens before this patch.

@roland-d i have no solution for that. I propose to not put a solution for taht in this PR and solve it in another PR.

infograf768 commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

roland-d commented:
@andrepereiradasilva No problem to put it in another PR. I won't be a common issue but yet still a bug.

roland-d commented:
@andrepereiradasilva Can you please fix the merge conflicts?

roland-d commented:
@andrepereiradasilva 
> Also remember you're testing and using a LOT of times. IMHO a regular user will not change it so many times.

Yes, I understand that but even doing 2 or 3 changes it jumps. Would be nice if we could just replace the content as it is.

roland-d commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<br /><br />I have checked all the permissions before and after. After applying the patch all the ACL changes are as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

roland-d commented:
Setting to RTC as we have 2 succesful tests<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

zero-24 commented:
@andrepereiradasilva there are merge conflicts.

andrepereiradasilva committed:
fix conflicts 1

andrepereiradasilva committed:
fix ocnflicts 1

andrepereiradasilva committed:
Merge remote-tracking branch 'refs/remotes/joomla/staging' into improve-ajax-js-errors

rishiv3 commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<br /><br />I have tested this item ✅ successfully on [126f797](https://github.com/joomla/joomla-cms/commit/126f79705c0f1ee6f0bdb2652a8176a507ca85e4 "126f797")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

andrepereiradasilva committed:
fix conflicts final

andrepereiradasilva commented:
@rishiv3 you tested in a not fully completed version.

Please test again

andrepereiradasilva committed:
solve bugs an add some debugging information

andrepereiradasilva commented:
@roland-d @infograf768 i added a temporary (or not) debug system.

Please enable debug. And check if everything is ok.

andrepereiradasilva committed:
cs

andrepereiradasilva committed:
cs + comments

andrepereiradasilva committed:
should be ===

andrepereiradasilva committed:
solve bugs and improve labels (hardcoded for now)

andrepereiradasilva committed:
small bugs

andrepereiradasilva committed:
solve roland issue (when the asset does not exist shouldn't create a asset it parent_id and level = 0)

andrepereiradasilva committed:
comments cs

andrepereiradasilva committed:
remove temporary debug information

andrepereiradasilva committed:
add new language strings

andrepereiradasilva committed:
solve minor bug in label of Public global config

roland-d commented:
I did the final testing once more and all looks good, all issues I found before are solved.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

rishiv3 commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<br /><br />tested again and everything looks good.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

zero-24 commented:
@roland-d can we RTC than?

gunjanpatel commented:
@zero-24 it is already in RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

roland-d commented:
If you guys @zero-24 and @gunjanpatel could test this as well it would be appreciated. It will be good to have some extra tests.

infograf768 commented:
Final testing done here. All looks real fine! Thanks @andrepereiradasilva  and @roland-d

andrepereiradasilva committed:
better message when trying to change permission in an itemwithout id (on create a new item)

andrepereiradasilva commented:
@infograf768 done as requested

brianteeman commented on administrator/language/en-GB/en-GB.com_config.ini:
```diff
@@ -230,11 +230,6 @@ COM_CONFIG_SAVE_SUCCESS="Configuration successfully saved."
-COM_CONFIG_SENDMAIL_JS_ERROR_NO_CONTENT="No content was returned."
-COM_CONFIG_SENDMAIL_JS_ERROR_OTHER="An error has occured while fetching the JSON data: HTTP %s status code."
-COM_CONFIG_SENDMAIL_JS_ERROR_PARSE="A parse error has occured while processing the following JSON data:<br/><code style="_QQ_"color:inherit;white-space:pre-wrap;padding:0;margin:0;border:0;background:inherit;"_QQ_">%s</code>"
-COM_CONFIG_SENDMAIL_JS_ERROR_TIMEOUT="A timeout has occured while fetching the JSON data."
```
Sorry to be a pain but the current rule is that strings are NOT deleted as they might be used by extensions and for another stupid reason not worthy of repeating

andrepereiradasilva replied:
no problem. i removed because they didn't exist in 3.5.1 ;)

See https://github.com/joomla/joomla-cms/blob/3.5.1/administrator/language/en-GB/en-GB.com_config.ini#L228-L237

brianteeman replied:
If they have never been in a release then its ok for them to be removed.
But if they have been in a release then they must stay

On 17 June 2016 at 11:09, andrepereiradasilva <notifications@github.com>
wrote:

> In administrator/language/en-GB/en-GB.com_config.ini
> <https://github.com/joomla/joomla-cms/pull/10793#discussion_r67488199>:
>
> > @@ -230,11 +230,6 @@ COM_CONFIG_SAVE_SUCCESS="Configuration successfully saved."
> >  COM_CONFIG_SENDMAIL_ACTION_BUTTON="Send Test Mail"
> >  COM_CONFIG_SENDMAIL_BODY="This is a test mail sent using "_QQ_"%s"_QQ_". If you receive it, then your email settings are correct!"
> >  COM_CONFIG_SENDMAIL_ERROR="Test mail could not be sent."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_CONNECTION_ABORT="A connection abort has occured while fetching the JSON data."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_NO_CONTENT="No content was returned."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_OTHER="An error has occured while fetching the JSON data: HTTP %s status code."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_PARSE="A parse error has occured while processing the following JSON data:<br/><code style="_QQ_"color:inherit;white-space:pre-wrap;padding:0;margin:0;border:0;background:inherit;"_QQ_">%s</code>"
> > -COM_CONFIG_SENDMAIL_JS_ERROR_TIMEOUT="A timeout has occured while fetching the JSON data."
>
> no problem. i removed because they didn't exist in 3.5.1 ;)
>
> See
> https://github.com/joomla/joomla-cms/blob/3.5.1/administrator/language/en-GB/en-GB.com_config.ini#L228-L237
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10793/files/5d72352c6d2c415e7f6864f2aeb6154cd8f8f77b#r67488199>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABPH8SArWwWJhbRxrPk7g6neiGS0uRYQks5qMnJXgaJpZM4Izlue>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

andrepereiradasilva replied:
they were not in any release.

They were added by me in april 7 
See https://github.com/andrepereiradasilva/joomla-cms/commit/cf3b3049d97068021196e2313010577dad505000

3.5.1 was released on Apr 5

brianteeman replied:
that's ok then

On 17 June 2016 at 11:21, andrepereiradasilva <notifications@github.com>
wrote:

> In administrator/language/en-GB/en-GB.com_config.ini
> <https://github.com/joomla/joomla-cms/pull/10793#discussion_r67489761>:
>
> > @@ -230,11 +230,6 @@ COM_CONFIG_SAVE_SUCCESS="Configuration successfully saved."
> >  COM_CONFIG_SENDMAIL_ACTION_BUTTON="Send Test Mail"
> >  COM_CONFIG_SENDMAIL_BODY="This is a test mail sent using "_QQ_"%s"_QQ_". If you receive it, then your email settings are correct!"
> >  COM_CONFIG_SENDMAIL_ERROR="Test mail could not be sent."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_CONNECTION_ABORT="A connection abort has occured while fetching the JSON data."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_NO_CONTENT="No content was returned."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_OTHER="An error has occured while fetching the JSON data: HTTP %s status code."
> > -COM_CONFIG_SENDMAIL_JS_ERROR_PARSE="A parse error has occured while processing the following JSON data:<br/><code style="_QQ_"color:inherit;white-space:pre-wrap;padding:0;margin:0;border:0;background:inherit;"_QQ_">%s</code>"
> > -COM_CONFIG_SENDMAIL_JS_ERROR_TIMEOUT="A timeout has occured while fetching the JSON data."
>
> they were not in any release.
>
> They were added by me in april 7
> See andrepereiradasilva@cf3b304
> <https://github.com/andrepereiradasilva/joomla-cms/commit/cf3b3049d97068021196e2313010577dad505000>
>
> 3.5.1 was released on Apr 5
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10793/files/5d72352c6d2c415e7f6864f2aeb6154cd8f8f77b#r67489761>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABPH8XwMGxkAE2RLScGE6WoeGLZvwu4Oks5qMnUngaJpZM4Izlue>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

brianteeman commented on administrator/language/en-GB/en-GB.lib_joomla.ini:
```diff
@@ -659,14 +666,18 @@ JLIB_RULES_INHERIT="Inherit"
+JLIB_RULES_NOT_ALLOWED_INHERITED="Not Allowed (Inherited)"
 JLIB_RULES_NOT_ALLOWED_LOCKED="Not Allowed (Locked)"
 JLIB_RULES_NOT_SET="Not Set"
+JLIB_RULES_NOTICE_RECALCULATE_GROUP_PERMISSIONS="Super user permissions changed. Save or reload to recalculate this group permissions."
```
Shouldnt this be Super User - capitalised

andrepereiradasilva replied:
done

brianteeman commented on administrator/language/en-GB/en-GB.lib_joomla.ini:
```diff
@@ -679,6 +690,9 @@ JLIB_UPDATER_ERROR_OPEN_UPDATE_SITE="Update: Could not open update site #%d &quo
 JLIB_USER_ERROR_AUTHENTICATION_LIBRARIES="JAuthentication: :__construct: Could not load authentication libraries."
 JLIB_USER_ERROR_BIND_ARRAY="Unable to bind array to user object."
+JLIB_USER_ERROR_CANNOT_CHANGE_SUPER_USER="A user is not allowed to change permissions of a Super User group."
+JLIB_USER_ERROR_CANNOT_CHANGE_OWN_GROUPS="A user is not allowed to change permissions of his own group(s)."
```
Please change to "their own" so it is gender neutral

andrepereiradasilva replied:
done

brianteeman commented on administrator/language/en-GB/en-GB.lib_joomla.ini:
```diff
@@ -679,6 +690,9 @@ JLIB_UPDATER_ERROR_OPEN_UPDATE_SITE="Update: Could not open update site #%d &quo
 JLIB_USER_ERROR_BIND_ARRAY="Unable to bind array to user object."
+JLIB_USER_ERROR_CANNOT_CHANGE_SUPER_USER="A user is not allowed to change permissions of a Super User group."
+JLIB_USER_ERROR_CANNOT_CHANGE_OWN_GROUPS="A user is not allowed to change permissions of his own group(s)."
+JLIB_USER_ERROR_CANNOT_CHANGE_OWN_PARENT_GROUPS="A user is not allowed to change permissions of his own group(s) parent group(s)."
```
Please change to "their own" so it is gender neutral

andrepereiradasilva replied:
done

brianteeman commented on language/en-GB/en-GB.lib_joomla.ini:
```diff
@@ -659,14 +666,18 @@ JLIB_RULES_INHERIT="Inherit"
+JLIB_RULES_NOT_ALLOWED_INHERITED="Not Allowed (Inherited)"
 JLIB_RULES_NOT_ALLOWED_LOCKED="Not Allowed (Locked)"
 JLIB_RULES_NOT_SET="Not Set"
+JLIB_RULES_NOTICE_RECALCULATE_GROUP_PERMISSIONS="Super user permissions changed. Save or reload to recalculate this group permissions."
```
As above Super User

andrepereiradasilva replied:
done

brianteeman commented on language/en-GB/en-GB.lib_joomla.ini:
```diff
@@ -679,6 +690,9 @@ JLIB_UPDATER_ERROR_OPEN_UPDATE_SITE="Update: Could not open update site #%d &quo
 JLIB_USER_ERROR_AUTHENTICATION_LIBRARIES="JAuthentication: :__construct: Could not load authentication libraries."
 JLIB_USER_ERROR_BIND_ARRAY="Unable to bind array to user object."
+JLIB_USER_ERROR_CANNOT_CHANGE_SUPER_USER="A user is not allowed to change permissions of a Super User group."
+JLIB_USER_ERROR_CANNOT_CHANGE_OWN_GROUPS="A user is not allowed to change permissions of his own group(s)."
```
Please change to "their own" so it is gender neutral

andrepereiradasilva replied:
done

brianteeman commented on language/en-GB/en-GB.lib_joomla.ini:
```diff
@@ -679,6 +690,9 @@ JLIB_UPDATER_ERROR_OPEN_UPDATE_SITE="Update: Could not open update site #%d &quo
 JLIB_USER_ERROR_BIND_ARRAY="Unable to bind array to user object."
+JLIB_USER_ERROR_CANNOT_CHANGE_SUPER_USER="A user is not allowed to change permissions of a Super User group."
+JLIB_USER_ERROR_CANNOT_CHANGE_OWN_GROUPS="A user is not allowed to change permissions of his own group(s)."
+JLIB_USER_ERROR_CANNOT_CHANGE_OWN_PARENT_GROUPS="A user is not allowed to change permissions of his own group(s) parent group(s)."
```
Please change to "their own" so it is gender neutral

andrepereiradasilva replied:
done

brianteeman commented on libraries/joomla/form/fields/rules.php:
```diff
@@ -319,75 +309,85 @@ protected function getInput()
+						$result['class'] = 'label label-important';
+						$result['text']  = '<span class="icon-lock icon-white"></span>' . JText::_('JLIB_RULES_NOT_ALLOWED_LOCKED');
+					}
+					// Some parent group as a explicit "Denied". Calculated permission is "Not Allowed (Locked)".
```
is that supposed to be "has an explicit"

andrepereiradasilva replied:
done

brianteeman commented on media/system/js/core-uncompressed.js:
```diff
@@ -218,6 +218,55 @@ Joomla.editors.instances = Joomla.editors.instances || {};
 
 	/**
+	 * Treat AJAX jQuery errors.
+	 * Used by some javascripts such as sentestamail.js and permissions.js
```
sendtestmail.js not sentestamail.js

andrepereiradasilva replied:
missed this one. will change.

andrepereiradasilva replied:
done

andrepereiradasilva committed:
eb-GB language corrections. thanks Brian

andrepereiradasilva committed:
sendtestmail.js not sentestamail.js

andrepereiradasilva commented:
folks, just language and comments changes as @brianteeman requested. no need to retest.

infograf768 commented:
ok. issue when new item corrected and strings modified for better english
@wilsonge 
at last, this can be merged. :+1:

andrepereiradasilva commented:
:sweat:

dgt41 commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

YGomiero commented:
I have tested this item ✅ successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

MATsxm commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<br /><br />Everything's ok here even if already RTC - Thanks all<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

mbabker commented:
Just one note, you don't need the permissions JS file with all 3 variations here.  It should either be .js and .min.js or -uncompressed.js and .js.  I honestly don't think JHtml's loader can cope with that correctly.

andrepereiradasilva commented:
there are only the normal and the uncompressed in this PR. (the .min.js was removed).
to follow the same logic of other js files in this folder.

See https://github.com/joomla/joomla-cms/tree/staging/media/system/js/

mbabker commented:
That's what I get for looking at the diff on mobile.

andrepereiradasilva commented:
is ok. some problem with your mobile them :smile: .

wilsonge commented:
Thanks @andrepereiradasilva and also thanks to @infograf768 and @roland-d who I know also put a lot of work into this!

andrepereiradasilva commented:
thanks you all! for testing and helping to solve this issues.

Aidan38 commented:
I have tested this item :white_check_mark: successfully on 126f79705c0f1ee6f0bdb2652a8176a507ca85e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10793">issues.joomla.org/joomla-cms/10793</a>.</sub>

ggppdk commented:
This is great work but I have found 2 issues

1 is serious and (was introduced by this PR), custom forms that do not have a form field named "title", can no longer save ACL permissions via AJAX, shall i open a new issue ?

andrepereiradasilva commented:
But please open an issue.
